pytest-embedded-idf/pytest_embedded_idf/dut.py

```diff
@@ -527,6 +365,211 @@ def run_all_single_board_cases(
                     elif case.type == 'multi_stage':
                         self._run_multi_stage_case(case, reset=reset, timeout=timeout)
 
+
+class IdfDut(IdfUnityDutMixin, SerialDut):
+    """
+    Dut class for serial ports connect to Espressif boards which are flashed with ESP-IDF apps
+
+    Attributes:
+        target (str): target chip type
+        skip_check_coredump (bool): skip check core dumped or not while dut teardown if set to True
+    """
+
+    XTENSA_TARGETS = ['esp32', 'esp32s2', 'esp32s3']
+    RISCV32_TARGETS = ['esp32c3', 'esp32h2', 'esp32c2']
```

Comment:
c6 as well?

Reply:
sure. also support detect with `IDF_TARGET_ARCH_XTENSA` and `IDF_TARGET_ARCH_RISCV`